lib/improver/utilities/temporal_interpolation.py

```diff
@@ -0,0 +1,419 @@
+                Array of sine of solar elevation at each point
+
+        """
+        day_of_year = (dtval - datetime(dtval.year, 1, 1)).days
+        utc_hour = (dtval.hour * 60.0 + dtval.minute) / 60.0
+        sin_phi = calc_solar_elevation(lats, lons, day_of_year,
+                                       utc_hour, return_sine=True)
+        return sin_phi
+
+    @staticmethod
+    def calc_lats_lons(cube):
+        """
+        Calculate the lats and lons of each point for a given cube.
+
+        Args:
+            cube (iris.cube.Cube):
+                cube containing x and y axis
+        Returns:
+            (tuple) : tuple containing:
+                **lats** (np.array):
+                    2d Array of latitudes for each point.
+                **lons** (np.array):
+                    2d Array of longitudes for each point.
+
+        """
+        trg_crs = lat_lon_determine(cube)
+        if trg_crs is not None:
+            xycube = next(cube.slices([cube.coord(axis='y'),
+                                       cube.coord(axis='x')]))
+            lats, lons = transform_grid_to_lat_lon(xycube)
+        else:
+            lats_row = cube.coord('latitude').points
+            lons_col = cube.coord('longitude').points
+            lats = np.repeat(lats_row[:, np.newaxis], len(lons_col), axis=1)
+            lons = np.repeat(lons_col[np.newaxis, :], len(lats_row), axis=0)
+        return lats, lons
+
+    def solar_interpolate(self, cube, interpolated_cube):
+        """
+        Interpolate solar radiation parameter which are zero if the
+        sun is below the horizon.
+
+        Args:
+            cube (iris.cube.Cube):
+                cube containing diagnostic cube valid at the beginning
+                of the period and cube valid at the end of the period
+            interpoldated_cube (iris.cube.Cube):
+                cube containing Linear interpolation of
+                cube at times in time_list.
+        Returns:
+            interpolated_cubes (iris.cube.CubeList):
+                A list of cubes interpolated to the desired times.
+
+        """
+
+        interpolated_cubes = iris.cube.CubeList()
+        (lats, lons) = self.calc_lats_lons(cube)
+        prev_data = cube[0].data
+        next_data = cube[1].data
+        dtvals = iris_time_to_datetime(cube.coord('time'))
+        dtval_prev = dtvals[0]
+        sin_phi_prev = self.calc_sin_phi(dtval_prev, lats, lons)
+        dtval_next = dtvals[1]
+        sin_phi_next = self.calc_sin_phi(dtval_next, lats, lons)
+        diff_step = (dtval_next - dtval_prev).seconds
+
+        for single_time in interpolated_cube.slices_over('time'):
+            # new_cubelist = iris.cube.CubeList()
+            dtval_interp = iris_time_to_datetime(single_time.coord('time'))[0]
+            sin_phi_interp = self.calc_sin_phi(dtval_interp, lats, lons)
+            diff_interp = (dtval_interp - dtval_prev).seconds
+            single_time.data[::] = 0.0
```

Comment:
Could you add some comments in this loop please about what's going on?  Eg here, I assume we're clearing all the data in order to reset the values with those interpolated using the solar method.  But it's not obvious if you're not fluent with numpy slice assignment.

Reply:
Comments added.